wled00/colors.cpp

```diff
@@ -91,6 +91,106 @@ void setRandomColor(byte* rgb)
+  palettecolors[keepcolorposition].hue += random8(20)-10; // +/- 10 randomness
+  //generate 4 saturation and brightness value numbers
+  //only one saturation is allowed to be below 200 creating mostly vibrant colors
+  //only one brigthness value number is allowed to below 200, creating mostly bright palettes  
+
+  for (int i = 0; i<3; i++) { //generate three high values
+    palettecolors[i].saturation = random8(180,255);
+    palettecolors[i].value = random8(180,255);
+  }
+  //allow one to be lower
+  palettecolors[3].saturation = random8(80,255);
+  palettecolors[3].value = random8(50,255);
+
+  //shuffle the arrays using Fisher-Yates algorithm
+  for (int i = 3; i > 0; i--) {
+    uint8_t j = random8(0, i + 1);
+    //swap [i] and [j]
+    uint8_t temp = palettecolors[i].saturation;
+    palettecolors[i].saturation = palettecolors[j].saturation;
+    palettecolors[j].saturation = temp;
+  }
+
+  for (int i = 3; i > 0; i--) {
+    uint8_t j = random8(0, i + 1);
+    //swap [i] and [j]
+    uint8_t temp = palettecolors[i].value;
+    palettecolors[i].value = palettecolors[j].value;
+    palettecolors[j].value = temp;
+  }
+
+  //now generate three new hues based off of the hue of the chosen current color
+  uint8_t basehue = palettecolors[keepcolorposition].hue;
+  uint8_t harmonics[3]; //hues that are harmonic but still a littl random
+  uint8_t type = random8(5); //choose a harmony type
+
+  switch (type) {
+    case 0: // analogous
+      harmonics[0] = basehue + random8(30, 50);
+      harmonics[1] = basehue + random8(10, 30);
+      harmonics[2] = basehue - random8(10, 30);
+      break;
+
+    case 1: // triadic
+      harmonics[0] = basehue + 110 + random8(20);
+      harmonics[1] = basehue + 230 + random8(20);
+      harmonics[2] = basehue + random8(30)-15;
+      break;
+
+    case 2: // split-complementary
+      harmonics[0] = basehue + 140 + random8(20);
+      harmonics[1] = basehue + 200 + random8(20);
+      harmonics[2] = basehue + random8(30)-15;
+      break;
+
+    case 3: // tetradic
+      harmonics[0] = basehue + 80 + random8(20);
+      harmonics[1] = basehue + 170 + random8(20);
+      harmonics[2] = basehue + random8(30)-15;
+      break;
+
+    case 4: // square
+      harmonics[0] = basehue + 80 + random8(20);
+      harmonics[1] = basehue + 170 + random8(20);
+      harmonics[2] = basehue + 260 + random8(20);
+      break;
+  }
+
+  //shuffle the hues:
+  for (int i = 2; i > 0; i--) {
+    uint8_t j = random8(0, i + 1);
+    //swap [i] and [j]
+    uint8_t temp = harmonics[i];
+    harmonics[i] = harmonics[j];
+    harmonics[j] = temp;
+  }
+
+  //now set the hues
+  int j=0;
+  for (int i = 0; i<4; i++) {
+    if(i==keepcolorposition) continue; //skip the base color
+    palettecolors[i].hue = harmonics[j];
+    j++;
+   }
+
+  return CRGBPalette16( palettecolors[0],
```

Comment:
Currently our palettes are assumed to be gamma-corrected.  You can use gamma8(uint8_t) or gamma32(uint32_t) to achieve that.